Source/ACE.Database/WorldDatabase.cs

```diff
@@ -73,6 +73,113 @@ public uint GetMaxGuidFoundInRange(uint min, uint max)
             }
         }
 
+        private static ConcurrentDictionary<uint, Weenie> level1Scrolls = new ConcurrentDictionary<uint, Weenie>();
+        private static ConcurrentDictionary<uint, Weenie> level2Scrolls = new ConcurrentDictionary<uint, Weenie>();
+        private static ConcurrentDictionary<uint, Weenie> level3Scrolls = new ConcurrentDictionary<uint, Weenie>();
+        private static ConcurrentDictionary<uint, Weenie> level4Scrolls = new ConcurrentDictionary<uint, Weenie>();
+        private static ConcurrentDictionary<uint, Weenie> level5Scrolls = new ConcurrentDictionary<uint, Weenie>();
+        private static ConcurrentDictionary<uint, Weenie> level6Scrolls = new ConcurrentDictionary<uint, Weenie>();
+
+        public Weenie GetSpellWeenie(uint tier, uint spellID)
```

Comment:
I think this should take a parameter like "worldIsPreloaded". If worldIsPreloaded, you wouldn't actually query the database, but instead would query the cached weenies.